docs/DEVELOPER_GUIDE.md

```diff
@@ -1 +1,173 @@
-todo
\ No newline at end of file
+# Development
+
+This doc explains how to setup a development environment so you can get started
+[contributing](../CONTRIBUTING.md)
+to `kfserving`. Also take a look at:
+
+- [How to add and run tests](../test/README.md)
+- [Iterating](#iterating)
+
+## Prerequisites
+
+Follow the instructions below to set up your development environment. Once you
+meet these requirements, you can make changes and
+[deploy your own version of kfserving](#deploy-kfserving)!
+
+Before submitting a PR, see also [CONTRIBUTING.md](../CONTRIBUTING.md).
+
+### Sign up for GitHub
+
+Start by creating [a GitHub account](https://github.com/join), then setup
+[GitHub access via SSH](https://help.github.com/articles/connecting-to-github-with-ssh/).
+
+### Install requirements
+
+You must install these tools:
+
+1. [`go`](https://golang.org/doc/install): KFServing controller is written in Go
+1. [`git`](https://help.github.com/articles/set-up-git/): For source control
+1. [`dep`](https://github.com/golang/dep): For managing external Go
+   dependencies.
+1. [`ko`](https://github.com/google/ko):
+   For development.
+1. [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/): For
+   managing development environments.
+
+### Install Knative on a Kubernetes cluster
+
+1. [Set up a kubernetes cluster and install Knative](https://knative.dev/docs/install/)
```

Comment:
possibly clarify that only knative-serving is required and knative-monitoring is good to have (and that others are currently not required?)

Reply:
I mean that you dont need to install the rest..

Reply:
Essentially mention to install 'Knative Serving' only.

Reply:
SGTM